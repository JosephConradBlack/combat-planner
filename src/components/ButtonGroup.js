import React from 'react';
import styled from 'styled-components';
import { Button, Icon, Divider } from 'semantic-ui-react';

const ButtonGroup = () => (
  <div>
    <Button basic>Clear Fighters</Button>
    <Button positive icon labelPosition="right" floated="right">
      Next Fighter<Icon name="right arrow" />
    </Button>
  </div>
);

export default ButtonGroup;
