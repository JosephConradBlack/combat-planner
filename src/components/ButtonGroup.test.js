import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ButtonGroup from './FighterList';

describe('ButtonGroup', () => {
  it('renders with no props', () => {
    const component = shallow(<ButtonGroup />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders with empty items array', () => {
    const component = shallow(<ButtonGroup items={[]} />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders with one items ', () => {
    const component = shallow(<ButtonGroup items={[{ id: 'id' }]} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
