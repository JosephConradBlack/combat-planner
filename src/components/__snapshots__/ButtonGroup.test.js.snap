// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup renders with empty items array 1`] = `
<Styled(SegmentGroup)>
  <PoseGroup
    enterPose="enter"
    exitPose="exit"
    flipMove={true}
    preEnterPose="exit"
    singleChildOnly={false}
  >
    <Component
      key="0"
    >
      <Styled(Fighter)
        key="0"
      />
    </Component>
    <Component
      key="1"
    >
      <Styled(Fighter)
        key="1"
      />
    </Component>
    <Component
      key="2"
    >
      <Styled(Fighter)
        key="2"
      />
    </Component>
  </PoseGroup>
</Styled(SegmentGroup)>
`;

exports[`ButtonGroup renders with no props 1`] = `
<Styled(SegmentGroup)>
  <PoseGroup
    enterPose="enter"
    exitPose="exit"
    flipMove={true}
    preEnterPose="exit"
    singleChildOnly={false}
  >
    <Component
      key="0"
    >
      <Styled(Fighter)
        key="0"
      />
    </Component>
    <Component
      key="1"
    >
      <Styled(Fighter)
        key="1"
      />
    </Component>
    <Component
      key="2"
    >
      <Styled(Fighter)
        key="2"
      />
    </Component>
  </PoseGroup>
</Styled(SegmentGroup)>
`;

exports[`ButtonGroup renders with one items  1`] = `
<Styled(SegmentGroup)>
  <PoseGroup
    enterPose="enter"
    exitPose="exit"
    flipMove={true}
    preEnterPose="exit"
    singleChildOnly={false}
  >
    <Component
      key="0"
    >
      <Styled(Fighter)
        key="0"
      />
    </Component>
    <Component
      key="1"
    >
      <Styled(Fighter)
        key="1"
      />
    </Component>
    <Component>
      <Styled(Fighter)
        index={2}
        item={Object {}}
        lastIndex={2}
      />
    </Component>
  </PoseGroup>
</Styled(SegmentGroup)>
`;
